src/spyglass/utils/mixins/ingestion.py

```diff
@@ -348,11 +353,36 @@ def validate_duplicates(self, entry_dict: Dict[dj.Table, List[dict]]):
         entry_dict : dict or Dict[dj.Table, List[dict]]
             The new entry or dict of table entries to validate against existing
             entries in the database.
+
+        Returns
+        -------
+        dict or Dict[dj.Table, List[dict]]
+            The new entries to insert after validation. Avoids need to flag
+            skip_duplicates
         """
+        entries_to_insert = dict()
         for table, table_entries in entry_dict.items():
-            for entry in self._adjust_keys_for_entry(table_entries):
+            if isinstance(table, type):
+                table = table()  # instantiate table object if class provided
+
+            entries_to_insert[table] = []
+            for table_entry in table_entries:
+                adjusted_entries = table._adjust_keys_for_entry([table_entry])
+                if not adjusted_entries:
+                    continue  # entry filtered out by adjustment
+                entry = adjusted_entries[0]
+                primary_entry = {
+                    k: v for k, v in entry.items() if k in table.primary_key
+                }
+                if not (table & primary_entry):
+                    # New entry, safe to insert
+                    entries_to_insert[table].append(table_entry)
```

Comment:
Just to confirm - this obj does not need to contain entries where the user accepts divergence because we're not inserting those, we're skipping in favor of the existing entries